src/components/Card.js

```diff
@@ -20,27 +34,83 @@ export default class Card {
     this._cardImage = this._cardElement.querySelector(".card__image");
     this._cardTitle = this._cardElement.querySelector(".card__title");
     this._likeButton = this._cardElement.querySelector(".card__like-button");
-    this._trashButton = this._cardElement.querySelector(".card__trash-button");
+    this._deleteButton = this._cardElement.querySelector(
+      ".card__delete-button"
+    );
     this._cardImage.src = this._link;
     this._cardImage.alt = this._name;
     this._cardTitle.textContent = this._name;
+    //this._cardElement.id = this.cardId;
     this._setEventListeners();
+    this.renderLikes();
+    this._handleLikeClick();
     return this._cardElement;
   }
-
   _handleLikeButton() {
-    this._likeButton.classList.toggle("card__like-button_active");
+    this._handleCardLike({ _id: this._id, likes: this._likes });
+    this.renderLikes();
+    this._handleLikeClick(this._id);
   }
-  _handleTrashButton() {
-    this._cardElement.remove();
+  toggleLikes() {
+    if (this.isLiked()) {
+      this._cardElement
+        .querySelector(".card__like-button")
+        .classList.add("card__like-button_active");
+    }
   }
   _setEventListeners() {
     this._cardImage.addEventListener("click", () =>
       this._handleCardClick({ name: this._name, link: this._link })
     );
     this._likeButton.addEventListener("click", () => this._handleLikeButton());
-    this._trashButton.addEventListener("click", () =>
-      this._handleTrashButton()
+    this._deleteButton.addEventListener("click", () =>
+      this._handleDeleteClick()
     );
+    this._checkUserId();
+  }
+  _getData() {
+    const data = {
+      name: this._name,
+      link: this._link,
+    };
+    return data;
+  }
+
+  getId() {
+    return this._id;
+  }
+
+  isLiked() {
+    return this._likes.some((like) => {
+      return like._id === this._myId;
+    });
+  }
+
+  setLikes(likes) {
+    this._likes = likes;
+    this.renderLikes();
+  }
+
+  renderLikes() {
+    this._cardElement.querySelector(".card__like-counter").textContent =
```

Comment:
`this._cardElement.querySelector(".card__like-counter")`

**Could be improved**


All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` and so on.